PWGHF/Utils/utilsEvSelHf.h

```diff
@@ -280,20 +282,59 @@ struct HfEventSelectionMc {
         zPvPosMin = option.defaultValue.get<float>();
       } else if (option.name.compare("hfEvSel.zPvPosMax") == 0) {
         zPvPosMax = option.defaultValue.get<float>();
+      } else if (option.name.compare("hfEvSel.centralityMin") == 0) {
+        centralityMin = option.defaultValue.get<float>();
+      } else if (option.name.compare("hfEvSel.centralityMax") == 0) {
+        centralityMax = option.defaultValue.get<float>();
       }
     }
   }
 
   /// \brief Function to apply event selections to generated MC collisions
   /// \param mcCollision MC collision to test against the selection criteria
+  /// \param collSlice collection of reconstructed collisions
+  /// \param centrality centrality variable to be set in this function
   /// \return a bitmask with the event selections not satisfied by the analysed collision
-  template <typename TBc, typename TMcColl>
-  uint16_t getHfMcCollisionRejectionMask(TMcColl const& mcCollision)
+  template <typename TBc, o2::hf_centrality::CentralityEstimator centEstimator, typename CCs, typename TMcColl>
+  uint16_t getHfMcCollisionRejectionMask(TMcColl const& mcCollision, CCs const& collSlice, float& centrality)
   {
     uint16_t rejectionMask{0};
     float zPv = mcCollision.posZ();
     auto bc = mcCollision.template bc_as<TBc>();
 
+    float multiplicity{0.f};
+    for (const auto& collision : collSlice) {
+      float collCent{0.f};
+      float collMult{0.f};
+      if constexpr (centEstimator != o2::hf_centrality::CentralityEstimator::None) {
+        if constexpr (centEstimator == o2::hf_centrality::CentralityEstimator::FT0A) {
+          collCent = collision.centFT0A();
+          collMult = collision.multFT0A();
+        } else if constexpr (centEstimator == o2::hf_centrality::CentralityEstimator::FT0C) {
+          collCent = collision.centFT0C();
+          collMult = collision.multFT0C();
+        } else if constexpr (centEstimator == o2::hf_centrality::CentralityEstimator::FT0M) {
+          collCent = collision.centFT0M();
+          collMult = collision.multFT0M();
+        } else if constexpr (centEstimator == o2::hf_centrality::CentralityEstimator::FV0A) {
+          collCent = collision.centFV0A();
+          collMult = collision.multFV0A();
+        } else {
+          LOGP(fatal, "Unsupported centrality estimator!");
+        }
+        if (collMult > multiplicity) {
+          centrality = collCent;
+          multiplicity = collMult;
+        }
```

Comment:
I would just use number of PV contributors (`collision.numContrib()`); the idea of selecting the collision with highest multiplicity is mainly because of split vertices (and typically the one with larger number of contributors is the good one).